config/storage/middleware/stores/patching.json

```diff
@@ -31,6 +32,64 @@
         }
       }
     },
+    {
+      "@id": "urn:solid-server:default:PatchHandler_RDFStore",
+      "@type": "WaterfallHandler",
+      "handlers": [
+        { "@id": "urn:solid-server:default:PatchHandler_ImmutableMetadata"},
+        { "@id": "urn:solid-server:default:PatchHandler_RDF" }
+      ]
+    },
+    {
+      "comment": "Patches metadata resources. Prevents specific triple patterns from being updated.",
+      "@id": "urn:solid-server:default:PatchHandler_ImmutableMetadata",
+      "@type": "ImmutableMetadataPatcher",
+      "patcher": { "@id": "urn:solid-server:default:PatchHandler_RDF" },
+      "metadataStrategy": { "@id": "urn:solid-server:default:MetadataStrategy" },
+      "immutablePatterns": [
+        {
+          "comment": "Pattern corresponding to the root storage of a Solid Pod.",
+          "@type": "FilterPattern",
+          "predicate": "http://www.w3.org/1999/02/22-rdf-syntax-ns#type",
+          "object": "http://www.w3.org/ns/pim/space#Storage"
+        },
+        {
+          "comment": "Pattern corresponding to the containment of resources.",
+          "@type": "FilterPattern",
+          "predicate": "http://www.w3.org/ns/ldp#contains"
+        },
+        {
+          "comment": "Pattern corresponding to describing the location of the metadata resource.",
+          "@type": "FilterPattern",
+          "predicate": "http://purl.org/dc/terms/description"
```

Comment:
The main problem is that if we allow other object values here it would be difficult for a client to determine where the metadata is located as there would be several potentially matching headers.

Reply:
Thinking on this again, we don't actually need to filter this one out.

@woutslabbinck instead of using this predicate to indicate what the location is a of metadata resource, we can just introduce a new `MetadataWriter` that generates the metadata URL using the auxiliary strategy instead of adding the entry to the `LinkRelMetadataWriter`.

Reply:
Thinking even more on these. Internally we should interpret these as always having as subject the target identifier, making them more specific.

Reply:
With `these`, do you mean all the filterPatterns? If so, this filterPattern can stay just the way it is, right?
```json
        {
          "comment": "The association of a resource and its metadata is managed by the server.",
          "@type": "FilterPattern",
          "predicate": "http://purl.org/dc/terms/description"
        },
```

Reply:
Just internally in the immutable patcher yes.

This pattern should still disappear since `dc:description` is a very generic predicate that people might want to use for something else as well, so we should still go with the metadata writer solution for this.